src/Serializer/ItemNormalizer.php

```diff
@@ -37,11 +39,17 @@ public function normalize($object, $format = null, array $context = [])
 
     /**
      * {@inheritdoc}
+     *
+     * @throws InvalidArgumentException
      */
     public function denormalize($data, $class, $format = null, array $context = [])
     {
         // Avoid issues with proxies if we populated the object
         if (isset($data['id']) && !isset($context['object_to_populate'])) {
+            if (isset($context['allow_update']) && true !== $context['allow_update']) {
```

Comment:
IMHO it's too late to make the changes. I spent several hours trying to make them with grep to find and sort (Symfony vs API Platform) the keys but I'm afraid about oversights and side effects.
I especially had some problems with Behat tests on serialization groups and I'm not so familiar with internals to fix them quickly.
Also, I think it could be problematic with users who currently use the different keys.
So for now, I only made the change for the `allow_update` key.

cc @dunglas